x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/aggregation/GroupingAggregatorFunctionTestCase.java

```diff
@@ -108,13 +110,41 @@ private Operator.OperatorFactory simpleWithMode(
         if (randomBoolean()) {
             supplier = chunkGroups(emitChunkSize, supplier);
         }
-        return new RandomizingHashAggregationOperatorFactory(
-            List.of(new BlockHash.GroupSpec(0, ElementType.LONG)),
-            mode,
-            List.of(supplier.groupingAggregatorFactory(mode, channels(mode))),
-            randomPageSize(),
-            null
-        );
+
+        if (shouldRandomizeBlocks()) {
+            return new RandomizingHashAggregationOperatorFactory(
+                List.of(new BlockHash.GroupSpec(0, ElementType.LONG)),
+                mode,
+                List.of(supplier.groupingAggregatorFactory(mode, channels(mode))),
+                randomPageSize(),
+                null
+            );
+        } else {
+            return new HashAggregationOperator.HashAggregationOperatorFactory(
+                List.of(new BlockHash.GroupSpec(0, ElementType.LONG)),
+                mode,
+                List.of(supplier.groupingAggregatorFactory(mode, channels(mode))),
+                randomPageSize(),
+                null
+            );
+        }
+    }
+
+    /**
+     * Checks whether the class is being run by {@link org.elasticsearch.compute.test.OperatorTestCase#testSimpleCircuitBreaking} or not.
+     * <p>
+     *     If it is, then we can't randomize blocks, as it expects a deterministic memory usage.
+     * </p>
+     * <p>
+     *     Explicitly linking {@link org.elasticsearch.compute.test.OperatorTestCase#testSimpleCircuitBreaking} as
+     *     this method looks for it based on its name, so any modification there should be updated here!
+     * </p>
+     */
+    private boolean shouldRandomizeBlocks() {
+        return Arrays.stream(Thread.currentThread().getStackTrace())
+            .noneMatch(
+                e -> e.getClassName().equals(OperatorTestCase.class.getName()) && e.getMethodName().equals("testSimpleCircuitBreaking")
```

Comment:
I wish we had C#'s `nameof()` to make this more robust. But I think adding that link in the javadoc is the only _simple_ option we have (?)